### PR TITLE
⚡️ Limit related courses on other pages

### PR DESCRIPTION
**Pull request by mbenadda**
## Purpose

Per #946.

Richie currently has scaling problems where large numbers of related items automatically loaded on some pages can cause them to be very slow. Eg. a category page on a Richie instance with ~500 courses can take up to 30 seconds to load.

This is unacceptable. Users probably don't even want to see hundred-items-long lists of courses or other objects on a related page.

## Proposal

- [x] Add settings/defaults/a context manager to limit the lengths of automatic lists of related objects
- [x] Create a filter to slice queries for related items through this new setting+context manager
- [x] Move `get_es_id` from categories indexer to category model so we can more easily build search URLs that filter by category
- [x] Ditto for organizations `get_es_id`
- [x] Limit the number of related courses on category pages
- [x] ... on organization pages
- [x] ... on person pages

### CODE REVIEW

File: src/richie/apps/courses/templates/courses/cms/category_detail.html [hunk 1]
```diff
@@ -92,6 +92,17 @@ <h2 class="course-glimpse-list__title">{% trans "Related courses" %}</h2>
           {% include "courses/cms/fragment_course_glimpse.html" %}
         {% endif %}
       {% endfor %}
+      {% if courses.count == PAGES_SETTINGS.MAX_RELATED_COURSES_ON_DETAILS_PAGES %}
+        {% if category.get_meta_category %}
```
Comment (sampaccoud) [inline src/richie/apps/courses/templates/courses/cms/category_detail.html]: why not show the search page filtered on this category? It seems to be what you naturally want to see and not all sibling categories.

File: src/richie/apps/courses/templatetags/first_n.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+@register.filter("first_n")
+def first_n_filter(value, arg):
+    """
+    Picks the first N items in a list.
+    The reason this exists over stock "slice" is to allow using a variable to define
+    the number of items to pick.
+    """
+    return value[slice(0, arg)]
```
Comment (sampaccoud) [inline src/richie/apps/courses/templatetags/first_n.py]: I find it very complicated to add a context manager just for this (remember that all projects using Richie will have to add our context managers to their settings for it to work... so we should avoid adding new ones).
Maybe considering the setting as a default and using it directly here can make all this easier. If our use cases get more complex later we will think of another way but for now it seems good enough?
Something like:
```suggestion
@register.filter("limit_related")
def limit_related(value, arg=None):
    arg = arg or getattr(settings, "RICHIE_LIMIT_RELATED_OBJECTS", 20)
    return value[slice(0, arg)]
```

File: src/richie/apps/courses/templates/courses/cms/category_detail.html (hunk 1 above)
Comment (mbenadda) [reply src/richie/apps/courses/templates/courses/cms/category_detail.html]: I'm not sure what you're suggesting here. It seems to me the link is precisely pointing to `the search page filtered on this category`

Comment (sampaccoud) [reply src/richie/apps/courses/templates/courses/cms/category_detail.html]: ok! I just realized why you need to get the id of the meta to do that :-/

**Comment by mbenadda** [comment]
This is handled better by #992.

**Pull request closed without merging**